File: problems/2315-count-asterisks/solution.py
class Solution:
    def countAsterisks(self, s: str) -> int:
        count = 0
        bars = 0
        for char in s:
            if char == '|':
                bars += 1
            elif char == '*' and bars % 2 == 0:
                count += 1
        return count

File: problems/2315-count-asterisks/solution.cpp
class Solution {
public:
    int countAsterisks(string s) {
        int count = 0;
        bool inside = false;
        for (char c : s) {
            if (c == '|') {
                inside = !inside;
            } else if (c == '*' && !inside) {
                count++;
            }
        }
        return count;
    }
};